clippy_lints/src/non_expressive_names.rs

```diff
@@ -168,6 +168,16 @@ fn allowed_to_be_similar(interned_name: &str, list: &[&str]) -> bool {
         .any(|&name| interned_name.starts_with(name) || interned_name.ends_with(name))
 }
 
+fn avoiding_keyword_clash(maybe_keyword: &str) -> bool {
+    let rust_keywords = [
+        "as", "break", "const", "continue", "crate", "else", "enum", "extern", "false", "fn", "for", "if", "impl",
+        "in", "let", "loop", "match", "mod", "move", "mut", "pub", "ref", "return", "self", "Self", "static", "struct",
+        "super", "trait", "true", "type", "unsafe", "use", "where", "while", "async", "await", "dyn", "abstract",
+        "become", "box", "do", "final", "macro", "override", "priv", "typeof", "unsized", "virtual", "yield", "union",
+    ];
+    rust_keywords.iter().any(|kw| *kw == maybe_keyword)
```

Comment:
This could be 

```suggestion
    Ident::with_dummy_span(Symbol::intern(maybe_keyword)).is_used_keyword()
```

Reply:
Thanks, much better than hardcoding the list. To also catch `yield` etc. I'd even use `is_reserved()`.